### PR TITLE
Delimiter option

### PR DESCRIPTION
**Pull request by gewfy**
As discussed this adds the possibility to have a custom delimiter.

### CODE REVIEW

**Comment by gewfy** [comment]
Nice catch, makes sense! Will dig deeper into that.

**Comment by gewfy** [comment]
Updated according to your request

File: src/widget/transform.ts [hunk 1]
```diff
@@ -1,21 +1,29 @@
 import type { Widget } from "../types";
 
-export const getName = (name: string, capitalize: boolean) =>
-  capitalize ? toCapitalized(name) : name;
-
-export const getWidgetName = (widget: Widget, useLabel: boolean, capitalize: boolean): string =>
+export const getName = (name: string, capitalize: boolean, delimiter: string) =>
+  toDelimiter(capitalize ? toCapitalized(name) : name, delimiter);
```
Comment (gewfy) [inline src/widget/transform.ts:4]: @jsaari97 I've changed approach a bit here and made sure to replace any existing delimiters too. With this you really get uniform type names. E.g. `kitchenSink_typed_list_type_2_object` with default options and where `type_2_object` is an actual name, can now with...
```javascript 
{ label:true, capitalize: true, delimiter: "" }
```
...become `KitchenSinkTypedListType2Object`.

WDYS?

**Review by jsaari97 — CHANGES_REQUESTED** [review]
Great job! 🎉 

One small change I would make is to define the default delimiter as a constant, eg `DEFAULT_DELIMITER` in the `constants.ts` file :)

**Comment by gewfy** [comment]
Ofc! Done :)

**Comment by jsaari97** [comment]
@gewfy looks like you accidentally forgot to add the constant to the CLI option's default value :)

**Comment by gewfy** [comment]
Doh! Fixed :)

**Review by jsaari97 — APPROVED** [review]

**Pull request merged**